CI failure: CI — TypeScript error in MCP server (model vs models)

## CI Failure Report

**Workflow:** CI
**Branch:** main
**Commit:** 410f7d1fa61917a73083298e05b950471ae9294b
**Run:** https://github.com/Action-Llama/action-llama/actions/runs/23357774937

## Failure Summary

TypeScript compilation failed in the MCP server code due to attempting to access non-existent `model` property on `AgentConfig` type, which should be `models` (plural).

## Failed Job(s)

### test

**Error output:**
```
src/mcp/server.ts(247,37): error TS2551: Property 'model' does not exist on type 'AgentConfig'. Did you mean 'models'?
src/mcp/server.ts(247,63): error TS2551: Property 'model' does not exist on type 'AgentConfig'. Did you mean 'models'?
Process completed with exit code 2.
```

## Root Cause Analysis

The issue is in `src/mcp/server.ts` at line 247, where the code attempts to access `config.model?.provider` and `config.model?.model`. However, the `AgentConfig` interface (defined in `src/shared/config.ts`) only has a `models` property (plural), which is an array of `ModelConfig` objects.

**Current code (line 247):**
```typescript
lines.push(`Model: ${config.model?.provider}/${config.model?.model || "default"}`);
```

**AgentConfig interface:**
```typescript
export interface AgentConfig {
  // ...
  models: ModelConfig[];  // Note: plural 'models', not 'model'
  // ...
}
```

This mismatch was likely introduced during recent refactoring that changed the agent configuration from a single model to multiple models (see commit d0b904a: "feat: named models and automatic model fallback chains").

## Suggested Fix

Update `src/mcp/server.ts` line 247 to use the `models` array instead of the non-existent `model` property. Since `models` is an array, we should access the first model:

```diff
- lines.push(`Model: ${config.model?.provider}/${config.model?.model || "default"}`);
+ lines.push(`Model: ${config.models[0]?.provider}/${config.models[0]?.model || "default"}`);
```

Alternatively, if we want to display multiple models, we could show all of them:

```diff
- lines.push(`Model: ${config.model?.provider}/${config.model?.model || "default"}`);
+ lines.push(`Models: ${config.models.map(m => `${m.provider}/${m.model}`).join(", ")}`);
```

## Additional Context

This issue was introduced by recent commits that refactored the agent configuration structure:
- d0b904a: "feat: named models and automatic model fallback chains"
- 3dc2cfe: "feat: align agent config with Agent Skills specification"

The `AgentConfig` interface was correctly updated, but this usage in the MCP server was missed.

---
*This issue was automatically created by the gh-actions-responder agent.*

--- a/.changeset/fix-mcp-server-models-display.md
+++ b/.changeset/fix-mcp-server-models-display.md
@@ -1,0 +1,5 @@
+---
+"@action-llama/action-llama": patch
+---
+
+Fixed MCP server agent info display to show all configured models instead of failing with TypeScript error. The server now correctly accesses the `models` array property instead of the non-existent `model` property. Closes #196.

--- a/src/mcp/server.ts
+++ b/src/mcp/server.ts
@@ -244,7 +244,7 @@ export async function startMcpServer(opts: {
         if (config.description) lines.push(`Description: ${config.description}`);
         if (config.schedule) lines.push(`Schedule: ${config.schedule}`);
         if (config.webhooks?.length) lines.push(`Webhooks: ${config.webhooks.map((w) => `${w.source}:${(w.events || []).join(",")}`).join(", ")}`);
-        lines.push(`Model: ${config.model?.provider}/${config.model?.model || "default"}`);
+        lines.push(`Models: ${config.models.length > 0 ? config.models.map(m => `${m.provider}/${m.model}`).join(", ") : "none"}`);
         if (config.credentials?.length) lines.push(`Credentials: ${config.credentials.join(", ")}`);
         if (config.scale && config.scale > 1) lines.push(`Scale: ${config.scale}`);
         if (config.timeout) lines.push(`Timeout: ${config.timeout}s`);

